api/v1alpha1/jwt_types.go

```diff
@@ -54,7 +54,11 @@ type JWTProvider struct {
 	// The claim must be of type; string, int, double, bool. Array type claims are not supported
 	//
 	ClaimToHeaders []ClaimToHeader `json:"claimToHeaders,omitempty"`
-	// TODO: Add TBD JWT fields based on defined use cases.
+
+	// ExtractFrom defines different ways to extract the JWT token from HTTP request.
+	// If empty, it defaults to extract JWT token from the Authorization HTTP request header using Bearer schema
+	// or access_token from query parameters.
+	ExtractFrom *JWTExtractor `json:"extractFrom,omitempty"`
```

Comment:
needs a `optional` annotation

Reply:
add json tag `omitempty` should have the same behavior with `optional` annotation, should i still need to add this?
Reference: https://book.kubebuilder.io/reference/markers.html?highlight=omitempty#markers-for-configcode-generation

Reply:
I believe`optional` is also added in other place to help with API docs